content/src/content/jcr_root/apps/core/wcm/components/carousel/v1/carousel/clientlibs/site/js/carousel.js

```diff
@@ -274,6 +316,73 @@
             resetAutoplayInterval();
         }
 
+        /**
+         * Handles pause element click events
+         *
+         * @private
+         * @param {Object} event The click event
+         */
+        function onPauseClick(event) {
+            pause();
+        }
+
+        /**
+         * Handles play element click events
+         *
+         * @private
+         * @param {Object} event The click event
+         */
+        function onPlayClick() {
+            play();
+        }
+
+        /**
+         * Pauses the playing of the Carousel. Sets {@code Carousel#_paused} marker.
+         * Only relevant when autoplay is enabled
+         *
+         * @private
+         */
+        function pause() {
+            that._paused = true;
+            clearAutoplayInterval();
+
+            var pause = that._elements["pause"];
+            if (pause) {
+                setActionDisabled(pause);
+            }
+
+            var play = that._elements["play"];
+            if (play) {
+                setActionDisabled(play, false);
+            }
+        }
+
+        /**
+         * Enables the playing of the Carousel. Sets {@code Carousel#_paused} marker.
+         * Only relevant when autoplay is enabled
+         *
+         * @private
+         */
+        function play() {
+            that._paused = false;
+
+            // If the Carousel is hovered, don't begin auto transitioning until the next mouse leave event
+            var hovered = that._elements.self.parentElement.querySelector(":hover") === that._elements.self;
+            if (that._properties.autopauseDisabled || !hovered) {
+                resetAutoplayInterval();
+            }
+
+            var pause = that._elements["pause"];
+            if (pause) {
+                setActionDisabled(pause, false);
```

Comment:
I find this a bit confusing to read. When I play the carousel, I would expect the `pause` action to be disabled, which would translate into: `setActionDisabled(pause, true);`.

Reply:
oh I see.. When the carousel is playing, the `pause` action is enabled so that it can be clicked on.
I would add a comment to explain it.